rfcs/rfcOVID-0018.md

```diff
@@ -0,0 +1,105 @@
+# A feature to automatically "yield true" at the end of a file
+
+## Preamble
+
+    Author:  Curtis "Ovid" Poe <curtis.poe@gmail.com>
+    Sponsor:
+    ID:      OVID-0018
+    Status:  Draft
+
+## Abstract
+
+This RFC proposes a `yield_true` feature. When used, the current Perl _file_
+containing `use feature 'yield_true';` will automatically return a true value
+after successful compilation, eliminating the need for a "1" (or other true
+value) at the end of the file.
+
+## Motivation
+
+Eliminate the need for a true value at the end of a Perl file.
+
+## Rationale
+
+There's no need to have a true value be hard-coded in our files that we
+`use`. Further, newer programmers can get confused because sometimes code
+_doesn't_ end with a true value but nonetheless compiles just fine because
+_something_ in the code returned a true value and the code compiles as a
+side-effect.
+
+## Specification
+
+    use feature 'yield_true';
+
+Code using the above does not need to return a magic true value when compiled.
+
+If the module explicitly returns a false value, we should consider if this can
+be ignored (I suspect this is usually a bug). However, the module author
+_might_ have conditional logic to explicitly return a false value in case
+there's a missing dependency (or for some other reason). Discussion might be
+useful here.
```

Comment:
If the module author wants the module to fail to load, the proper thing to do is die with an actual error message.

Reply:
I've fixed that in the RFC.